### PR TITLE
Save layout of the Decompiler Widgets

### PR DESCRIPTION
**Pull request by NirmalManoj**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**
The decompiler widget was made sync-able and the support for multiple decompiler widgets was implemented by PR #2402. However, the layout management was not implemented in that PR, and because of this layouts were not restored. This PR aims to solve this issue.

Even though this PR will restore the layout, in case multiple widgets are open with multiple decompilers, all widgets will be opened with the last opened decompiler. The logic for implementing the ideal situation where the decompiler will also be restored will require complex refactorings. I don't intend to do that by this PR. This is not a big deal for now considering the fact that we are not restoring seek.

**Test plan (required)**
1. Compile PR and test if the layout gets restored as it should.
2. I'm not sure if there is anything else that I should add, so please have a quick look at how other memory widgets store and restore layouts and check if anything relevant is missing in this PR.

**Closing issues**

### CODE REVIEW

File: src/widgets/DecompilerWidget.cpp [hunk 1]
```diff
@@ -108,6 +108,11 @@ DecompilerWidget::DecompilerWidget(MainWindow *main) :
 
 DecompilerWidget::~DecompilerWidget() = default;
 
+QString DecompilerWidget::getWidgetType()
```
Comment (karliss) [inline src/widgets/DecompilerWidget.cpp:111]: It's probably better to use this when doing setObjectName so that they are in sync.

**Review by karliss — CHANGES_REQUESTED** [review]

File: src/widgets/DecompilerWidget.cpp (hunk 1 above)
Comment (karliss) [reply src/widgets/DecompilerWidget.cpp:111]: Also object name shouldn't be translated it's an internal ID not a UI string.

**Comment by karliss** [comment]
> The logic for implementing the ideal situation where the decompiler will also be restored will require complex refactorings. 

It shouldn't require a lot of refactoring. The mechanism for widget specific view setting saving is already there. All you have to do is implement `serializeViewProprties` and `deserializeViewProperties`.

File: src/widgets/DecompilerWidget.cpp (hunk 1 above)
Comment (NirmalManoj) [reply src/widgets/DecompilerWidget.cpp:111]: Thanks for pointing this out. Fixed it.

**Comment by NirmalManoj** [comment]
> It shouldn't require a lot of refactoring. The mechanism for widget specific view setting saving is already there. All you have to do is implement `serializeViewProprties` and `deserializeViewProperties`.

Thanks for telling me. I will look into this and will see how this can be done. But not in this PR.

**Review by karliss — APPROVED** [review]
Looks good. I am investigating the travis failure. Unlikely to be related to your issue but needs to be fixed either way.

**Comment by ITAYC0HEN** [comment]
how can we avoid such issues next times we add a widget? I would personally forget about it

**Comment by karliss** [comment]
a) it only affects widgets with multiple copies, for others it's more difficult to get wrong
b) refactoring to reduce amount of code that needs to be updated
c) developer docs

**Pull request merged**